test/e2e/expirable-airdrop.spec.ts

```diff
@@ -0,0 +1,302 @@
+
+      // approve
+      await token.approve(expirableAirdrop.address, ethers.constants.MaxUint256);
+
+      // deposit
+      await expirableAirdrop.depositTokens(toDeposit);
+    });
+
+    then('airdrop is deposited', async () => {
+      // expect
+      expect(await token.balanceOf(expirableAirdrop.address)).equal(toDeposit);
+    });
+
+    then('event is emitted', async () => {
+      // expect
+      expect(tx).to.have.emit(expirableAirdrop, 'Deposited').withArgs(toDeposit);
+    });
+  });
+
+  when('claimAndSendToClaimee()', () => {
+    let leaf: any;
+    let proof: any;
+
+    given(async () => {
+      await prepareAirdrop();
+
+      leaf = getLeaf(alice.address, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+    });
+
+    when('airdrop is claimed by claimee', () => {
+      let tx: Transaction;
+
+      given(async () => {
+        tx = await expirableAirdrop.connect(alice).claimAndSendToClaimee(alice.address, airdropAmountBN, proof);
+      });
+
+      then('tokens are transferred to claimee', async () => {
+        // expect
+        expect(await token.balanceOf(alice.address)).equal(airdropAmountBN);
+      });
+
+      then('event is emitted', async () => {
+        // expect
+        expect(tx).to.have.emit(expirableAirdrop, 'Claimed').withArgs(alice.address, alice.address, airdropAmountBN);
+      });
+
+      then('claimee is marked as claimed', async () => {
+        // expect
+        await expect(expirableAirdrop.connect(alice).claimAndSendToClaimee(alice.address, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+          expirableAirdrop,
+          'AlreadyClaimed'
+        );
+      });
+    });
+
+    when('airdrop is claimed on behalf of', () => {
+      let tx: Transaction;
+
+      given(async () => {
+        tx = await expirableAirdrop.connect(bob).claimAndSendToClaimee(alice.address, airdropAmountBN, proof);
+      });
+
+      then('tokens are transferred to claimee', async () => {
+        // expect
+        expect(await token.balanceOf(alice.address)).equal(airdropAmountBN);
+      });
+
+      then('event is emitted', async () => {
+        // expect
+        expect(tx).to.have.emit(expirableAirdrop, 'Claimed').withArgs(alice.address, alice.address, airdropAmountBN);
+      });
+
+      then('claimee is marked as claimed', async () => {
+        // expect
+        await expect(expirableAirdrop.connect(alice).claimAndSendToClaimee(alice.address, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+          expirableAirdrop,
+          'AlreadyClaimed'
+        );
+      });
+    });
+
+    then('revert if no airdrop', async () => {
+      leaf = getLeaf(randomAddress, airdropAmountBN);
+      proof = tree.getHexProof(leaf);
+
+      await expect(expirableAirdrop.connect(random).claimAndSendToClaimee(randomAddress, airdropAmountBN, proof)).to.be.revertedWithCustomError(
+        expirableAirdrop,
+        'NotInMerkle'
+      );
+    });
```

Comment:
```suggestion
   when('there is no airdrop', () => {
      then('tx gets reverted with custom error', async () => {
        leaf = getLeaf(randomAddress, airdropAmountBN);
        proof = tree.getHexProof(leaf);

        await expect(expirableAirdrop.connect(random).claimAndSendToClaimee(randomAddress, airdropAmountBN, proof)).to.be.revertedWithCustomError(
          expirableAirdrop,
          'NotInMerkle'
        );
      });
   });
```

I'd rather have this naming and `describe, given, when then`  🚀 . Applies for the same test below.